src/Lower.cpp

```diff
@@ -218,7 +218,7 @@ Module lower(const vector<Function> &output_funcs, const string &pipeline_name,
     if (t.has_gpu_feature() ||
         t.has_feature(Target::OpenGLCompute) ||
         t.has_feature(Target::OpenGL) ||
-        (t.arch != Target::Hexagon && (t.features_any_of({Target::HVX_64, Target::HVX_128})))) {
+        (t.arch != Target::Hexagon && (t.features_any_of({Target::HVX_64, Target::HVX_128, Target::HexagonDma})))) {
```

Comment:
I think you just want to treat HexagonDma as a GPU feature (in this if, equivalent to e.g. has_gpu_feature(), Target::OpenGL, or Target::OpenGLCompute), so it should be outside the check for t.arch != Target::Hexagon.